lib/streamlit/state/session_state.py

```diff
@@ -68,27 +71,58 @@ class Value:
     value: Any
 
 
-WState = Union[Serialized, Value]
+WState = Union[Value, Serialized]
 
 WidgetArgs = Tuple[Any, ...]
+WidgetKwargs = Dict[str, Any]
 WidgetCallback = Callable[..., None]
+
 # A deserializer receives the value from whatever field is set on the
 # WidgetState proto, and returns a regular python value. A serializer
 # receives a regular python value, and returns something suitable for
 # a value field on WidgetState proto. They should be inverses.
-WidgetDeserializer = Callable[[Any, str], Any]
-WidgetSerializer = Callable[[Any], Any]
-WidgetKwargs = Dict[str, Any]
+WidgetDeserializer = Callable[[Any, str], T]
+WidgetSerializer = Callable[[T], Any]
+
+ArrayValueType = Literal[
+    "double_array_value",
+    "int_array_value",
+    "string_array_value",
+]
+
+ValueType = Literal[
+    ArrayValueType,
+    "arrow_value",
+    "bool_value",
+    "bytes_value",
+    "double_value",
+    "file_uploader_state_value",
+    "int_value",
+    "json_value",
+    "string_value",
+    "trigger_value",
+]
+
+ARRAY_VALUE_TYPES: Final = frozenset(
+    cast(
+        Set[ArrayValueType],
+        {
+            "double_array_value",
+            "int_array_value",
+            "string_array_value",
+        },
+    )
+)
```

Comment:
Can you add comments above `ArrayValueType`, `ValueType`, and `ARRAY_VALUE_TYPES` indicating what they're for? (I think it's appropriate that they're pulled to the top of the source file, but it also makes their meaning less obvious than it was when they were inline.)

Reply:
I made a couple of changes, inspired by your comment. I added comments, like you asked for, but I also moved these types to `type_util`, in order to make this file more compact and easier to read/scroll, and because they are useful in multiple files.

Also changed the type names to end with `...FieldName`, rather than `...Type`, to better reflect what these strings "really are". This is also preferable because `...Type` carries a lot of connotations in python, particularly that the objects should be usable as an instance constructor, which is not the case here.